### ISSUE TITLE
The app crashes when you try to open the dropin

### ISSUE THREAD

**Issue by eloy98g** [Adyen/adyen-react-native#96]
Hi, we are having a problem when using the component <AdyenCheckout>.  The application crashes when pressing the button with the `start("dropIn")` action. The expected behaviour is that the "dropin" opens

The error log:
`com.facebook.react.bridge.ReadableNativeArray cannot be cast tocom.facebook.react.bridge.ReadableNativeMap`

Code: 
```
<AdyenCheckout
            config={configuration}
            paymentMethods={paymentMethods}
            onSubmit={(payload, nativeComponent) =>
              submitHandler(payload, nativeComponent)
            }
            makePaymentsCall={(payload, nativeComponent) =>
              submitHandler(payload, nativeComponent)
            }
            onProvide={provideHandler}
            onFail={failHandler}
            onComplete={onComplete}
          >
            <Button
              title="Open DropIn"
              onPress={() => {
                start("dropIn");
              }}
            />
 </AdyenCheckout>
```

Note: `configuration` is the configuration object with the same format as in the package's example
`paymentMethods`: `[
  {
    "brands": [
      "visa",
      "mc",
    ],
    "name": "Tarjeta de Crédito",
    "type": "scheme",
  },
   {
    "brands": [
      "mc",
      "visa",
    ],
    "configuration": {
      "merchantId": ...,
      "merchantName": "Gopick",
    },
    "name": "Apple Pay",
    "type": "applepay",
  },
   {
    "configuration": {
      "gatewayMerchantId": ...,
      "merchantId": ...,
    },
    "name": "Google Pay",
    "type": "googlepay",
  },
]` 

We are compiling with eas.
expo version is: "43.0.2",

We are blocked by this problem, what could be the reasons for this?

**Comment by descorp** [comment]
Hey @eloy98g 

`paymentMethods` [supposed to be an object](https://github.com/Adyen/adyen-react-native/blob/develop/ios/AdyenModule.m#L13), since it consist of 2 arrays: `paymentMethods` and `storedPaymentMethods`.
See example of [paymentMethods](https://docs.adyen.com/api-explorer/Checkout/68/post/paymentMethods) response.
We recommend to pass the response directly to `AdyenCheckout` - no extra actions needed.
 
I believe you should be able to fix your code by wrapping your `paymentMethods` array in object:

```js
<AdyenCheckout
            ...
            paymentMethods={{ paymentMethods:  paymentMethods }}
            ...
>
```

---

BTW, code snippet above might not work 🤔
You are calling `start("dropIn")` inside of the context, but I believe that you are creating `start` outside of the cortex - this could lead `start` to be unidentified.

If you face above issue - check samples on [Readme](https://github.com/Adyen/adyen-react-native#opening-payment-component).

**Comment by descorp** [comment]
Hey @eloy98g 

Any luck?

**Comment by descorp** [comment]
I am going to close this thread due to lack of activity

**Comment by Seamus1989** [comment]
same issue here. Followed the relatively basic installation instructions, getting a crash on IOS

**Comment by descorp** [comment]
Hey @Seamus1989 

If you have the same `com.facebook.react.bridge.ReadableNativeArray cannot be cast to com.facebook.react.bridge.ReadableNativeMap` error - this means something that must be **Dictionary** - is **Array**.

Are you using [paymentMethods](https://docs.adyen.com/api-explorer/Checkout/68/post/paymentMethods) from backend response or creating it manually?

**Comment by Seamus1989** [comment]
Well for testing purposes I just have it hardcoded in for now, but it is the same response each time (I'm just setting up and getting started). Looks something like this...

```
{
        paymentMethods: [
          {
            configuration: {
              merchantId: 'some_id',
              gatewayMerchantId: 'some_id',
            },
            name: 'Google Pay',
            type: 'googlepay',
          },
          { brands: ['mc', 'visa'], name: 'Credit Card', type: 'scheme' },
        ],
      }
```
and then using this like so..

```
  return (
    <AdyenCheckout
      config={configuration}
      paymentMethods={paymentMethods}
```
I should also add I am using typescript, and this is happy with this. The crash occurs as soon as `start()` is called.

**Comment by Seamus1989** [comment]
The example App also crashes [here](https://github.com/Adyen/adyen-react-native/tree/develop/example)

**Comment by descorp** [comment]
@Seamus1989 

Our SDK provides Type description for [PaymentMethods](https://github.com/Adyen/adyen-react-native/blob/develop/src/Core/types.ts#L175-L185). Object should have following structure:
```js
const manualPaymentMethods = {
    paymentMethods: [
      { brands: ['mc', 'visa'], name: 'Credit Card', type: 'scheme', }, 
      . . .
    ],
    storedPaymentMethods: [ ] // optional
}

<AdyenCheckout
      paymentMethods={manualPaymentMethods}
    . . .
</AdyenCheckout     
```

---

This confusion can be avoided by use of Typescript or by enabling `// @ts-check`.

**Comment by Seamus1989** [comment]
that is exactly what I have

**Comment by descorp** [comment]
@Seamus1989 

In this case, your problem is caused by something else.
Could you create new issue and provide full crashlog from React Native and from Xcode ?
